### DISCUSSION TITLE
how to pretty print an angle degree with `°` instead of `deg`

### DISCUSSION THREAD

**Discussion by kompre** [hgrecco/pint#1942] [Q&A]
Is it possible to pretty print a degree angle with the `°` symbol?

This example:

```python
from pint import UnitRegistry

u = UnitRegistry()

f'{1.0 * u.deg:~P}'
```

will return `'1.0 deg'`; I would prefer to be instead `'1.0 °'`.

Also if I set the locale to something, it will return nothing

```python
u.set_fmt_locale('en_US')
f'{1.0 * u.deg:~P}'
```

it will return `'1.0 '`

**Comment by hgrecco** [comment]
You can make a copy of this file

https://github.com/hgrecco/pint/blob/master/pint/default_en.txt

and then replace `deg` by your desired symbol.

and the load this file `u = UnitRegistry("myfile.txt")`

**Reply by kompre** [reply]
Thanks. I assume the command needs to run everytime I initiate a unit registry?

Is there a "safe" way to make it a permanent config of my installation?

**Reply by hgrecco** [reply]
>  I assume the command needs to run everytime I initiate a unit registry?
Indeed. But it not really a different command. `u = UnitRegistry()` means use the definitions file provided by the package

> Is there a "safe" way to make it a permanent config of my installation?
Not really. What I have done in the past is create a small package and import the regitry from there

**Comment by andrewgsavage** [comment]
We could change the DefaultRegistry to look for a default_en.txt file in
the user_data_dir and use it, if it exists
